tests/tensor/test_nlinalg.py

```diff
@@ -580,3 +581,42 @@ def test_eval(self):
         t_binv1 = tf_b1(self.b1)
         assert _allclose(t_binv, n_binv)
         assert _allclose(t_binv1, n_binv1)
+
+class TestKron(utt.InferShapeTester):
+    rng = np.random.default_rng(43)
+
+    def setup_method(self):
+        self.op = kron
+        super().setup_method()
+
+    def test_perform(self):
+        for shp0 in [(2,), (2, 3), (2, 3, 4), (2, 3, 4, 5)]:
+            x = tensor(dtype="floatX", shape=(None,) * len(shp0))
+            a = np.asarray(self.rng.random(shp0)).astype(config.floatX)
+            for shp1 in [(6,), (6, 7), (6, 7, 8), (6, 7, 8, 9)]:
+                if len(shp0) + len(shp1) == 2:
+                    continue
+                y = tensor(dtype="floatX", shape=(None,) * len(shp1))
+                f = function([x, y], kron(x, y))
+                b = self.rng.random(shp1).astype(config.floatX)
+                out = f(a, b)
+                # Newer versions of scipy want 4 dimensions at least,
+                # so we have to add a dimension to a and flatten the result.
+                if len(shp0) + len(shp1) == 3:
+                    scipy_val = scipy.linalg.kron(a[np.newaxis, :], b).flatten()
+                else:
+                    scipy_val = scipy.linalg.kron(a, b)
```

Comment:
change `scipy.linalg.kron` here to `np.kron`. Try and see if we can remove this if-else block as well, it seems to have something to do with the scipy implementation.